Source/Core/Common/bbaclient.h

```diff
@@ -0,0 +1,39 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <QObject>
+
+class QTcpSocket;
+
+class BbaServer;
+class BbaDebug;
+
+class BbaClient : public QObject
+{
+  Q_OBJECT
+
+public:
+  explicit BbaClient(QTcpSocket &socket, BbaServer &server);
+
+  BbaDebug logInfo();
+
+Q_SIGNALS:
```

Comment:
You can just use `signal` and `slot`, there's no need to use the macros here. (Q_SLOTS isn't needed with Qt 5 signal and slots anymore anyways).